pkg/install/stack/seed-mla/stack.go

```diff
@@ -627,3 +638,48 @@ func upgradeKubeStateMetricsDeployment(
 	}
 	return nil
 }
+
+func upgradeNodeExporterDaemonSets(
+	ctx context.Context,
+	logger *logrus.Entry,
+	kubeClient ctrlruntimeclient.Client,
+	helmClient helm.Client,
+	opt stack.DeployOptions,
+	chart *helm.Chart,
+	release *helm.Release,
+) error {
+	logger.Infof("%s: %s detected, performing upgrade to %s…", release.Name, release.Version.String(), chart.Version.String())
+	// 1: find the old daemonset
+	logger.Info("Backing up old Node Exporter daemonset…")
+
+	daemonsetsList := &unstructured.UnstructuredList{}
+	daemonsetsList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "apps",
+		Kind:    "DaemonSetList",
+		Version: "v1",
+	})
+
+	nodeExporterMatcher := ctrlruntimeclient.MatchingLabels{
+		"app": NodeExporterReleaseName,
+	}
+	if err := kubeClient.List(ctx, daemonsetsList, ctrlruntimeclient.InNamespace(NodeExporterNamespace), nodeExporterMatcher); err != nil {
+		return fmt.Errorf("error querying API for the existing DaemonSet object, aborting upgrade process.")
+	}
+
+	// 2: store the daemonset for backup
+	backupTS := time.Now().Format("2006-01-02T150405")
+	filename := fmt.Sprintf("backup_%s_%s.yaml", NodeExporterReleaseName, backupTS)
+	logger.Infof("Attempting to store the daemonsets in file: %s", filename)
+	if err := util.DumpResources(ctx, filename, daemonsetsList.Items); err != nil {
+		return fmt.Errorf("failed to back up the daemonsets, it is not removed: %w", err)
+	}
+
+	// 3: delete the daemonset
+	logger.Info("Deleting the daemonsets from the cluster")
+	for _, obj := range daemonsetsList.Items {
+		if err := kubeClient.Delete(ctx, &obj); err != nil {
+			return fmt.Errorf("failed to remove the daemonset: %w\n\nuse backup file to check the changes and restore if needed", err)
+		}
+	}
+	return nil
+}
```

Comment:
The PR, in general, looks really good. I'm just a bit skeptical about this migration after going through the upgrade docs for this change. We should remove the helm release instead of just the Daemonset to ensure that everything that `node-exporter` installs has been cleaned up. Just removing the Daemonset would leave redundant remanent resources.

 Also w.r.t https://github.com/kubermatic/docs/pull/1842#discussion_r2000950737, IMO we don't need any automated migration here. The user can manually delete the helm release and perform the ugprade.